hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SystemSetup.java

```diff
@@ -149,42 +156,67 @@ public void doPostUpgradeSetup(@NonNull final Dispatch dispatch) {
         // permissions from any upgrade files that are present in the configured directory
         final var filesConfig = config.getConfigData(FilesConfig.class);
         final var adminConfig = config.getConfigData(NetworkAdminConfig.class);
-        final List<AutoSysFileUpdate> autoUpdates = List.of(
-                new AutoSysFileUpdate(
-                        createFileID(filesConfig.feeSchedules(), config),
+        final List<AutoEntityUpdate<Bytes>> autoSysFileUpdates = List.of(
+                new AutoEntityUpdate<>(
+                        (ctx, bytes) ->
+                                dispatchSynthFileUpdate(ctx, createFileID(filesConfig.feeSchedules(), config), bytes),
                         adminConfig.upgradeFeeSchedulesFile(),
                         SystemSetup::parseFeeSchedules),
-                new AutoSysFileUpdate(
-                        createFileID(filesConfig.throttleDefinitions(), config),
+                new AutoEntityUpdate<>(
+                        (ctx, bytes) -> dispatchSynthFileUpdate(
+                                ctx, createFileID(filesConfig.throttleDefinitions(), config), bytes),
                         adminConfig.upgradeThrottlesFile(),
                         SystemSetup::parseThrottles),
-                new AutoSysFileUpdate(
-                        createFileID(filesConfig.networkProperties(), config),
+                new AutoEntityUpdate<>(
+                        (ctx, bytes) -> dispatchSynthFileUpdate(
+                                ctx, createFileID(filesConfig.networkProperties(), config), bytes),
                         adminConfig.upgradePropertyOverridesFile(),
                         in -> parseConfig("override network properties", in)),
-                new AutoSysFileUpdate(
-                        createFileID(filesConfig.hapiPermissions(), config),
+                new AutoEntityUpdate<>(
+                        (ctx, bytes) -> dispatchSynthFileUpdate(
+                                ctx, createFileID(filesConfig.hapiPermissions(), config), bytes),
                         adminConfig.upgradePermissionOverridesFile(),
                         in -> parseConfig("override HAPI permissions", in)));
-        autoUpdates.forEach(update -> {
+        autoSysFileUpdates.forEach(update -> {
             if (update.tryIfPresent(adminConfig.upgradeSysFilesLoc(), systemContext)) {
                 dispatch.stack().commitFullStack();
             }
         });
+        final var autoNodeAdminKeyUpdates = new AutoEntityUpdate<Map<Long, Key>>(
+                (ctx, nodeAdminKeys) -> nodeAdminKeys.forEach(
+                        (nodeId, key) -> ctx.dispatchAdmin(b -> b.nodeUpdate(NodeUpdateTransactionBody.newBuilder()
+                                .nodeId(nodeId)
+                                .adminKey(key)
+                                .build()))),
+                adminConfig.upgradeNodeAdminKeysFile(),
+                SystemSetup::parseNodeAdminKeys);
```

Comment:
This kind of adminKey update looks dangerous to me. If we have some key in this node-admin-keys.json file, the node adminKey will be updated. This should be guarded carefully.